web/router.ex

```diff
@@ -11,9 +11,13 @@ defmodule CodeCorps.Router do
 
   pipeline :api do
     plug :accepts, ["json-api", "json"]
+    plug JaSerializer.Deserializer
+  end
+
+  pipeline :with_token do
     plug Guardian.Plug.VerifyHeader
     plug Guardian.Plug.LoadResource
-    plug JaSerializer.Deserializer
+    plug CodeCorps.Plug.CurrentUser
```

Comment:
I think that instead of a custom plug, we can use `Guardian.Plug.EnsureAuthenticated`. This should cause the resource to be loaded (`guardian_serializer` module) as well as make sure a valid token exists.